tests/js/index.js

```diff
@@ -63,10 +63,24 @@ exports.runTest = function(suiteName, testName) {
         // Start fresh in case of a crash in a previous run.
         Realm.clearTestState();
 
+        var promise;
         try {
-            testMethod.call(testSuite);
+            promise = testMethod.call(testSuite);
+
+            // If the test returns a promise, then clear state on success or failure.
```

Comment:
why would the test not return a promise? wouldn't we still want to clear the state in that case?